operations/mimir-mixin/dashboards/dashboard-utils.libsonnet

```diff
@@ -1715,7 +1715,78 @@ local utils = import 'mixin-utils/utils.libsonnet';
         )
       ),
-    ],
+    ] +
+    $.ingesterStoreGatewayReadsDashboardsRows(
+      'ingester',
+      $.queries.ingester.requestsPerSecondMetric,
+      $.queries.ingester.readRequestsPerSecondSelector,
+      $._config.job_names.ingester,
+      $._config.ingester_read_path_routes_regex,
+      $.queries.querier.ingesterClientRequestsPerSecondMetric,
+      querierJobName,
+    ) +
+    $.ingesterStoreGatewayReadsDashboardsRows(
+      'store-gateway',
+      $.queries.store_gateway.requestsPerSecondMetric,
+      $.queries.store_gateway.readRequestsPerSecondSelector,
+      $._config.job_names.store_gateway,
+      $._config.store_gateway_read_path_routes_regex,
+      $.queries.querier.storeGatewayClientRequestsPerSecondMetric,
+      querierJobName,
+    ),
+
+  ingesterStoreGatewayReadsDashboardsRows(
+    ingesterStoreGatewayComponentName,
+    ingesterStoreGatewayRequestsPerSecondMetric,
+    ingesterStoreGatewayRequestsPerSecondSelector,
+    ingesterStoreGatewayJobNames,
+    ingesterStoreGatewayRoutesRegex,
+    querierRequestsPerSecondMetric,
+    querierJobNames,
+  ):: [
+    local description = 'This panel shows %(ingesterStoreGatewayComponentName)s query requests from all sources: the main query path, and the remote ruler query path, if in use. The data shown is as reported by %(ingesterStoreGatewayComponentName)ss.' % { ingesterStoreGatewayComponentName: ingesterStoreGatewayComponentName };
+
+    $.row($.capitalize(ingesterStoreGatewayComponentName + ' - query requests from all sources'))
+    .addPanel(
+      $.timeseriesPanel('Requests / sec') +
+      $.panelDescription('Requests / sec', description) +
+      $.qpsPanelNativeHistogram(ingesterStoreGatewayRequestsPerSecondMetric, ingesterStoreGatewayRequestsPerSecondSelector)
+    )
+    .addPanel(
+      $.timeseriesPanel('Latency') +
+      $.panelDescription('Latency', description) +
+      $.latencyRecordingRulePanelNativeHistogram(ingesterStoreGatewayRequestsPerSecondMetric, $.jobSelector(ingesterStoreGatewayJobNames) + [utils.selector.re('route', ingesterStoreGatewayRoutesRegex)])
+    )
+    .addPanel(
+      $.timeseriesPanel('Per %s %s p99 latency' % [ingesterStoreGatewayComponentName, $._config.per_instance_label]) +
+      $.panelDescription('Per %s %s p99 latency' % [ingesterStoreGatewayComponentName, $._config.per_instance_label], description) +
+      $.perInstanceLatencyPanelNativeHistogram(
+        '0.99',
+        ingesterStoreGatewayRequestsPerSecondMetric,
+        $.jobSelector(ingesterStoreGatewayJobNames) + [utils.selector.re('route', ingesterStoreGatewayRoutesRegex)],
+      ),
+    ),
+
+    local description = 'This panel shows %(ingesterStoreGatewayComponentName)s query requests from just this query path. The data shown is as reported by queriers.' % { ingesterStoreGatewayComponentName: ingesterStoreGatewayComponentName };
```

Comment:
I think we should take the path as a param to `ingesterStoreGatewayReadsDashboardsRows` and then describe what the path is rather than "this query path" which can be confusing.

Reply:
Good idea, done in 8159548.